src/agent/src/image_rpc.rs

```diff
@@ -121,10 +140,44 @@ impl ImageService {
         Ok(())
     }
 
+    // If we fail to start the AA, Skopeo/ocicrypt won't be able to unwrap keys
+    // and container decryption will fail.
+    fn init_attestation_agent() {
+        let config_path = OCICRYPT_CONFIG_PATH;
+
+        // The image will need to be encrypted using a keyprovider
+        // that has the same name (at least according to the config).
+        let ocicrypt_config = serde_json::json!({
+            "key-providers": {
+                "attestation-agent":{
+                    "grpc":AA_PORT
+                }
+            }
+        });
+
+        let mut config_file = fs::File::create(config_path).unwrap();
+        config_file
+            .write_all(ocicrypt_config.to_string().as_bytes())
+            .unwrap();
+
+        // The Attestation Agent will run for the duration of the guest.
+        Command::new(AA_PATH)
+            .arg("--grpc_sock")
+            .arg(AA_PORT)
+            .spawn()
+            .unwrap();
+
+        //TODO: Do we need to clean up the AA on shutdown?
+    }
+
     async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
         let image = req.get_image();
         let mut cid = req.get_container_id();
 
+        let agent_config = AGENT_CONFIG.read().await;
+        let aa_kbc_params = &agent_config.aa_kbc_params;
+        let aa_started = agent_config.confidential_setup_complete;
```

Comment:
`let aa_started = !agent_config.confidential_setup_complete;` ?
Or change the naming?

Reply:
If the setup is complete the agent has been started, so I think this name is fine. I changed the `!` below.

Reply:
Oh, yes,  my mistake,  naming is ok.